proxycurl/tools/profile.py

```diff
@@ -0,0 +1,40 @@
+from tools.helper import tool_registry, format_url, api_key_headers
+from tools.search import search_company, search_user
+from tools.api import school_profile_from_url, user_profile_from_url, company_profile_from_url
+import os
+
+
+@tool_registry.decorator("GetCompanyProfile")
+def get_company_profile():
+    env_url = os.getenv("URL")
+    company_name = os.getenv("COMPANY")
+    url = env_url if env_url else f'https://www.linkedin.com/company/{format_url(company_name)}/'
+
+    response = company_profile_from_url(url)
+
+    # Search for URL with company name
+    if response.status_code == 404 and company_name:
+        print("Initial request 404. Searching for profile URL...")
+
+        os.environ["NAME"] = company_name
+        return search_company()
```

Comment:
If you take my earlier suggestion, you can just call the search function here with a `max_results` of 1 and it should still behave the same way.